### ISSUE TITLE
Update to FCS 40

### ISSUE THREAD

**Issue by baronfel** [ctaggart/FsAst#36]
We're blocked from doing this until fantomas has a version that supports FCS 40. There's a branch with that work, but no prerelease/stable quite yet.

**Comment by Zaid-Ajaj** [comment]
Wouldn't be a good idea to keep FCS used here in sync with FCS version in Fantomas? Had problems using different versions

Heck, I even thought about including FsAst into Fantomas (maybe in a `.Writers` namespace) since they are always used together

**Comment by baronfel** [comment]
> Wouldn't be a good idea to keep FCS used here in sync with FCS version in Fantomas? Had problems using different versions

yes, that's precisely why the associated MR is in draft state still :p

**Comment by baronfel** [comment]
@nojaf speaking of, any idea when the first prereleases of fantomas with FCS40 will drop? I'm also looking forward to those for FSAC itself

**Comment by nojaf** [comment]
Hey, yeah about that 🙈. I'm currently pitching [the idea](https://github.com/dotnet/fsharp/issues/11801#issuecomment-880129676) of having a patch release for FCS 40. There is [one thing](https://github.com/dotnet/fsharp/issues/11732) that just really doesn't produce AST anymore. And [another fix](https://github.com/dotnet/fsharp/issues/11734) that would allow for a huge clean up in Fantomas.
I've poked @vzarytovskii about it and they would discuss it internally.

**Comment by baronfel** [comment]
the thing is, even that proposed patch release would be a major version bump (due to api breakages). so it sounds like you're suggesting skipping 40 entirely?

**Comment by nojaf** [comment]
It has crossed my mind, to be honest. The daemon experiment could potentially resolve the hurdle in FSAC.
And as for FsAst, I'm not sure what the impact would be. If there is any...

**Comment by nojaf** [comment]
https://www.nuget.org/packages/FSharp.Compiler.Service/40.0.1-preview.21352.5 just dropped, I'll check over the weekend if I can publish the first alpha.
